redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java

```diff
@@ -53,6 +63,36 @@ public DatabaseConfig register(DatabaseConfig configToSave) {
             }
             throw e;
         }
+    }
+
+    public DatabaseConfig delete(String name) {
+        try {
+            return transactionService.required(() -> {
+                Optional<DatabaseConfig> foundDatabaseConfig = databaseConfigRepository.findByName(name);
+                DatabaseConfig databaseConfig = foundDatabaseConfig
+                        .orElseThrow(() -> new NotFoundException(String.format("Database with name '%s' not found", name)));
+                if (databaseConfig.isUserManaged()) {
+                    deleteDatabase(databaseConfig);
+                }
+                delete(databaseConfig);
+                return databaseConfig;
+            });
+        } catch (TransactionExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof NotFoundException) {
+                throw (NotFoundException) cause;
+            }
+            LOGGER.warn("Deleting of DatabaseConfig failed: ", e);
+            throw new RuntimeException("transaction went fail");
```

Comment:
"transaction failed"

Also, this should include e as a cause, and then it doesn't have to log anything because code higher up the stack can handle it.

Reply:
Yes, thanks for that. I added e, very important.